tests/test_open_options.py

```diff
@@ -0,0 +1,32 @@
+"""Tests of dataset opening options and driver choice"""
+
+from packaging.version import parse
+import pytest
+
+import rasterio
+from rasterio.transform import Affine
+
+
+@pytest.mark.xfail(parse(rasterio.__gdal_version__) < parse('2.0'),
+                   reason="Requires a GDAL 2.0 feature")
+def test_fail_with_missing_driver():
+    """Fail to open a GeoTIFF without the GTiff driver"""
+    with pytest.raises(rasterio.errors.RasterioIOError):
+        rasterio.open('tests/data/RGB.byte.tif', driver='BMP')
+
+
+@pytest.mark.xfail(parse(rasterio.__gdal_version__) < parse('2.0'),
+                   reason="Requires a GDAL 2.0 feature")
+def test_open_specific_driver():
+    """Open a GeoTIFF with the GTiff driver"""
+    with rasterio.open('tests/data/RGB.byte.tif', driver='GTiff') as src:
+        assert src.count == 3
```

Comment:
We can open a GeoTIFF using the GTiff driver 🎉

Reply:
This is what we can do to select a specific JP2 driver.

Reply:
This is super minor, but this GDAL version dependent test uses `xfail` and the other version dependent test introduced in this PR uses `skipif`.  In broader terms, which one should we be using?  I'm inclined to use `skipif` rather than `xfail`.  These version dependent tests are *invalid* on specific GDAL versions, so its not so much we expect them to fail as they just do not apply to a specific condition.  From the [`pytest` docs](https://docs.pytest.org/en/latest/skipping.html):

> A skip means that you expect your test to pass only if some conditions are met, otherwise pytest should skip running the test altogether. Common examples are skipping windows-only tests on non-windows platforms, or skipping tests that depend on an external resource which is not available at the moment (for example a database).

>  A xfail means that you expect a test to fail for some reason. A common example is a test for a feature not yet implemented, or a bug not yet fixed. When a test passes despite being expected to fail (marked with pytest.mark.xfail), it’s an xpass and will be reported in the test summary.